### PR TITLE
Fix empty code parameter in CodeVerifierAuthenticator

### PR DESCRIPTION
**Pull request by aijazkeerio**
Fixes https://github.com/spring-projects/spring-authorization-server/issues/1671

### CODE REVIEW

**Review by 9527286768 — APPROVED** [review]

**Review by 9527286768 — APPROVED** [review]

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java [hunk 1]
```diff
@@ -140,7 +140,8 @@ private static boolean authorizationCodeGrant(Map<String, Object> parameters) {
 		// @formatter:off
 		return AuthorizationGrantType.AUTHORIZATION_CODE.getValue().equals(
 				parameters.get(OAuth2ParameterNames.GRANT_TYPE)) &&
-				parameters.get(OAuth2ParameterNames.CODE) != null;
+				parameters.get(OAuth2ParameterNames.CODE) != null &&
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java:143]: If the grant type is `authorization_code` and the `code` is empty, it should not return `false` and instead should error.

Try this instead:

```java
if (!AuthorizationGrantType.AUTHORIZATION_CODE.getValue().equals(parameters.get(OAuth2ParameterNames.GRANT_TYPE))) {
	return false;
}
if (!StringUtils.hasText((String) parameters.get(OAuth2ParameterNames.CODE))) {
	throwInvalidGrant(OAuth2ParameterNames.CODE);
}
return true;
```

File: oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProviderTests.java [hunk 2]
```diff
@@ -282,6 +283,29 @@ public void authenticateWhenAuthorizationCodeGrantAndValidCredentialsThenAuthent
 		assertThat(authenticationResult.getRegisteredClient()).isEqualTo(registeredClient);
 	}
 
+	@Test
+	public void authenticateWhenAuthorizationCodeGrantAndPkceAndValidCodeVerifierAndMissingCodeThenAuthenticated() {
```
Comment (jgrandja) [inline oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProviderTests.java:287]: Please remove this test and add `OAuth2AuthorizationCodeGrantTests.requestWhenPublicClientWithPkceAndEmptyCodeThenBadRequest()`. You can use `requestWhenPublicClientWithPkceThenReturnAccessTokenResponse()` as a template and adjust it.

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @aijazkeerio. Please see review comments.

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java (hunk 1 above)
Comment (aijazkeerio) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java]: Please review new changes.

**Comment by jgrandja** [comment]
Thanks for the updates @aijazkeerio ! This is now merged.

FYI, I added a polish commit that was mostly formatting changes.

**Pull request closed without merging**